Source/engine.cpp

```diff
@@ -49,6 +82,32 @@ BYTE *GetLightTable(char light) {
 
 } // namespace
 
+std::uint_fast16_t MeasureCelWidth(BYTE *pCelBuf) {

```

Comment:
Considering that we have all the cel widths hardcoded in the items them selfs, is this measuring the non-transparent part of the graphics, and not the cell itself?

Reply:
On a side note, this should be run when loading the item graphics and assigned to a new item property so that it doesn't run each frame.

Reply:
Yeah, we'll need to know the dimensions anyway when doing clipped rendering

Reply:
> is this measuring the non-transparent part of the graphics, and not the cell itself?

Pretty sure this is measuring the Cel width itself.

Reply:
I think what @qndel wanted as a function that measure the bounding box of the graphics in the cell. Some items can look quite off-center because of where in the sprite the graphics appear.

Reply:
Yeah I've understood that now. Such a function can be easily written, and @qndel's hack doesn't quite do that actually.